svc/FetchService.js

```diff
@@ -101,32 +162,16 @@ export class FetchService {
         return ret;
     }
 
-    /**
-     * Send an HTTP request to a URL, and decode the response as JSON.
-     *
-     * @param {Object} opts - options to pass through to fetch, with some additions.
-     *      @see https://developer.mozilla.org/en-US/docs/Web/API/Request for the available options
-     * @param {string} opts.url - target url to send the HTTP request to. Relative urls will be
-     *     appended to XH.baseUrl for the request
-     * @param {Object} [opts.params] - parameters to encode and send with the request body (for POSTs)
-     *      or append as a query string.
-     * @param {string} [opts.method] - The HTTP Request method to use for the request. If not
-     *     explicitly set in opts then the method will be set to POST if there are params,
-     *     otherwise it will be set to GET.
-     * @param {string} [opts.contentType] - value to use in the Content-Type header in the request.
-     *     If not explicitly set in opts then the contentType will be set based on the method. POST
-     *     requests will use 'application/x-www-form-urlencoded', otherwise 'text/plain' will be
-     *     used.
-     * @returns {Promise} the decoded JSON object, or null if the response had no content.
-     */
-    async fetchJson(opts) {
-        const ret = await this.fetch({acceptJson: true, ...opts});
-        return ret.status === 204 ? null : ret.json();
-    }
-
     //-----------------------
     // Implementation
     //-----------------------
+    async sendJson(opts) {
+        opts = Object.assign(opts, {
```

Comment:
Can use ... operator here.

Reply:
How would you do that without overwriting the stringify  of opts.body?

Reply:
true. done.